File: Math/Sum of Square Numbers.java
class Solution {
    public boolean judgeSquareSum(int c) {
        int hem=0,kar=(int)Math.sqrt(c);
        while(hem<=kar){
            long sum=(long)(hem*hem)+(long)(kar*kar);
            if(sum==c)return true;
            else if(sum>c)kar--;
            else hem++;
        }
        return false;
    }
}

File: Math/Sum of Square Numbers.py
import math
class Solution(object):
    def judgeSquareSum(self, c):
        a=0
        b=int(math.sqrt(c))
        while a<=b :
            sum=(a*a)+(b*b)
            if sum ==c:
                return True
            elif sum >c:
                b=b-1
            else :
                a=a+1
        return False

        
        
